### PR TITLE
refactor: rewrite project in vue and vite

### PR DESCRIPTION
**Pull request by boojack**
Tech stack: Vue3 + Vite + Vuex + TailwindCSS

### CODE REVIEW

**Comment by boojack** [comment]
PTAL

**Comment by tianzhou** [comment]
Overall logic LGTM. @xiaoluoboding please take a look at the code.

File: src/components/Dialog.vue [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<template>
+  <div
+    :class="
+      'fixed w-full px-3 md:px-0 h-full flex flex-col justify-center items-center bg-black bg-opacity-60 z-50 top-0 left-0 ' +
+      classname
```
Comment (xiaoluoboding) [inline src/components/Dialog.vue:5]: We need to distinguish static class from dynamic class

In Vue Project, we write like this:

```html
<div
  class="xxxx"
  :class="xxxx"
/>
```

File: src/components/Footer.vue [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+<script lang="ts">
+import { defineComponent } from "vue";
+
+export default defineComponent({
+  name: "Footer",
+  setup() {
+    return {};
+  },
+});
+</script>
```
Comment (xiaoluoboding) [inline src/components/Footer.vue:69]: There is no logic, only template.

File: src/components/Header.vue [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+      </div>
+      <div class="h-full flex md:hidden flex-row justify-end items-center">
+        <span
+          class="relative h-full w-10 px-3 flex flex-row justify-center items-center cursor-pointer font-semibold text-light hover:bg-zinc-800"
+          @click="handleToggleDropMenuBtnClick"
+        >
+          <span
+            :class="`w-4 transition-all h-px bg-light absolute top-1/2 ${
+              state.showDropMenu ? 'w-6 rotate-45' : '-mt-1'
+            }`"
```
Comment (xiaoluoboding) [inline src/components/Header.vue:51]: We need to distinguish static class from dynamic class.

File: src/components/RepoInputer.vue [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+    return {
+      state,
+      inputElRef,
+      isFetching: computed(() => {
+        return store.state.isFetching;
+      }),
+      repos: computed(() => {
+        return store.state.repos;
+      }),
```
Comment (xiaoluoboding) [inline src/components/RepoInputer.vue:173]: move the `computed` part below the `ref`, it's some state of something you want use.

keep `return` clean. maybe computed logic is too huge.

File: src/components/Toast.vue [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    textColor() {
+      switch (this.$props.type) {
+        case "normal":
+          return "text-white";
+        case "warn":
+          return "text-white";
+        case "succeed":
+          return "text-white";
+      }
+      return "text-white";
```
Comment (xiaoluoboding) [inline src/components/Toast.vue:84]: always `text-white` ?

Comment (xiaoluoboding) [inline src/components/Toast.vue:84]: always `text-white` ?

**Review by xiaoluoboding — CHANGES_REQUESTED** [review]

File: src/components/Dialog.vue (hunk 1 above)
Comment (boojack) [reply src/components/Dialog.vue]: Greate idea

**Comment by boojack** [comment]
@xiaoluoboding PTAL

**Review by xiaoluoboding — APPROVED** [review]

**Comment by xiaoluoboding** [comment]
LGTM

**Pull request merged**